lib/Crypto/Encryption.php

```diff
@@ -164,12 +164,14 @@ public function getDisplayName() {
 	 * @param string $mode php stream open mode
 	 * @param array $header contains the header data read from the file
 	 * @param array $accessList who has access to the file contains the key 'users' and 'public'
+	 * @param bool|string Either false or the name of source file to be renamed. This is helpful
+	 * 						for revision increment during move operation between storage.
 	 *
 	 * @return array $header contain data as key-value pairs which should be
 	 *                       written to the header, in case of a write operation
 	 *                       or if no additional data is needed return a empty array
 	 */
-	public function begin($path, $user, $mode, array $header, array $accessList) {
+	public function begin($path, $user, $mode, array $header, array $accessList, $renameFile = false) {
```

Comment:
Brought a new argument to begin. There are respective changes https://github.com/owncloud/core/pull/28107. Made changes to encryptions fopen. Also to the "ocencryption" context.